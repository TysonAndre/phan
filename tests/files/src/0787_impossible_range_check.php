<?php
function test787(string $str, bool $b, ?bool $nullable) {
    $value = rand() % 2 ? 3 : 4;
    var_export($value > 4);
    var_export('foo' >= $b);
    $s = rand(0,1) ? 'FOO' : 'bar';
    var_export($s >= $nullable);
}
test787('test', (bool)rand(0,1), null);
