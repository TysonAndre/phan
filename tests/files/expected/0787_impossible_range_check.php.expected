%s:4 PhanSuspiciousValueComparison Suspicious attempt to compare $value of type 3|4 to 4 of type 4 with operator '>'
%s:5 PhanSuspiciousValueComparison Suspicious attempt to compare 'foo' of type 'foo' to $b of type bool with operator '>='
%s:7 PhanSuspiciousValueComparison Suspicious attempt to compare $s of type 'FOO'|'bar' to $nullable of type ?bool with operator '>='
